File: Task 1_Chat-Bot/chat_bot.py
# Define a dictionary to store the rules and responses
rules = {
    "hello|hi|hey": "Hello! How can I assist you today?",
    "how are you|what's up": "I'm doing well, thanks! How about you?",
    "what's your name": "I'm a chatbot, nice to meet you!",
    "help|assist": "I can help with math, coding, and general questions. What do you need help with?",
    "goodbye|bye|see you": "Goodbye! It was nice chatting with you.",
    "default": "I didn't understand that. Can you please rephrase?"
}

def respond(user_input):
    # Convert the user input to lowercase for case-insensitive matching
    user_input = user_input.lower()

    # Iterate through the rules and check if the user input matches any of them
    for pattern, response in rules.items():
        if any(word in user_input for word in pattern.split("|")):
            return response

    # If no rule matches, return the default response
    return rules["default"]

# Test the chatbot
while True:
    user_input = input("You: ")
    response = respond(user_input)
    print("Chatbot:", response)

File: Task 1_Chat-Bot/Chatbot.java
import java.util.Scanner;

public class Chatbot {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String rules = "hello|hi|hey:Hello! How can I assist you today?;" +
                       "how are you|what's up:I'm doing well, thanks! How about you?;" +
                       "what's your name:I'm a chatbot, nice to meet you!;" +
                       "help|assist:I can help with math, coding, and general questions. What do you need help with?;" +
                       "goodbye|bye|see you:Goodbye! It was nice chatting with you.;" +
                       "default:I didn't understand that. Can you please rephrase?";

        String[] rulePairs = rules.split(";");

        while (true) {
            System.out.print("You: ");
            String userInput = scanner.nextLine().toLowerCase();

            boolean matched = false;
            for (String rulePair : rulePairs) {
                String[] parts = rulePair.split(":");
                String pattern = parts[0];
                String response = parts[1];

                String[] words = pattern.split("\\|");
                for (String word : words) {
                    if (userInput.contains(word)) {
                        System.out.println("Chatbot: " + response);
                        matched = true;
                        break;
                    }
                }

                if (matched) {
                    break;
                }
            }

            if (!matched) {
                System.out.println("Chatbot: " + rulePairs[rulePairs.length - 1].split(":")[1]);
            }
        }
    }
}